bindings/wasm/examples/src/1_advanced/0_did_controls_did.ts

```diff
@@ -0,0 +1,130 @@
+      }
+    ]
+  }
+
+  // Adding the issuer feature means we have to recalculate the required storage deposit.
+  subsidiaryAlias.amount = TransactionHelper.getStorageDeposit(subsidiaryAlias, rentStructure).toString();
+
+  // Publish the subsidiary's DID.
+  subsidiaryDocument = await didClient.publishDidOutput(secretManager, subsidiaryAlias);
+
+  // =====================================
+  // Update the subsidiary's Alias Output.
+  // =====================================
+
+  // Add a verification method to the subsidiary.
+  // This only serves as an example for updating the subsidiary DID.
+  const keyPair: KeyPair = new KeyPair(KeyType.Ed25519);
+  const method: IotaVerificationMethod = new IotaVerificationMethod(
+    subsidiaryDocument.id(),
+    KeyType.Ed25519,
+    keyPair.public(),
+    "#key-2"
+  );
+  subsidiaryDocument.insertMethod(method, MethodScope.VerificationMethod());
+
+  // Update the subsidiary's Alias Output with the updated document
+  // and increase the storage deposit.
+  const subsidiaryAliasUpdate: IAliasOutput = await didClient.updateDidOutput(subsidiaryDocument);
+  subsidiaryAliasUpdate.amount = TransactionHelper.getStorageDeposit(subsidiaryAliasUpdate, rentStructure).toString();
+
+  // Publish the updated subsidiary's DID.
+  //
+  // This works because `secret_manager` can unlock the company's Alias Output,
+  // which is required in order to update the subsidiary's Alias Output.
+  subsidiaryDocument = await didClient.publishDidOutput(secretManager, subsidiaryAliasUpdate);
+
+  // ===================================================================
+  // Determine the controlling company's DID given the subsidiary's DID.
+  // ===================================================================
+
+  // Resolve the subsidiary's Alias Output.
+  const subsidiaryOutput: IAliasOutput = await didClient.resolveDidOutput(subsidiaryDocument.id());
+
+  // Extract the company's Alias Id from the state controller unlock condition.
+  //
+  // If instead we wanted to determine the original creator of the DID,
+  // we could inspect the issuer feature. This feature needs to be set when creating the DID.
+  //
+  // Non-null assertion is safe to use since every Alias Output has a state controller unlock condition.
+  // Cast to IStateControllerAddressUnlockCondition is safe as we check the type in find.
+  const stateControllerUnlockCondition: IStateControllerAddressUnlockCondition =
+    subsidiaryOutput.unlockConditions.find(unlockCondition => unlockCondition.type == STATE_CONTROLLER_ADDRESS_UNLOCK_CONDITION_TYPE)! as IStateControllerAddressUnlockCondition
+
+  // Cast to IAliasAddress is safe because we set an Alias Address earlier.
+  const companyAliasId: string = (stateControllerUnlockCondition.address as IAliasAddress).aliasId;
+
+  // Reconstruct the company's DID from the Alias Id and the network.
+  companyDid = new IotaDID(Converter.hexToBytes(companyAliasId), networkName);
```

Comment:
Should we make an issue add an  `IotaDID.fromAliasId` function to the Wasm bindings, if this is a common operation?

Reply:
Yeah seems reasonable to add. Opened #1025.